blog/_posts/2021-04-08-scala-3-in-sbt.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: blog-detail
+post-type: blog
+by: Adrien Piquerez, Scala Center
+title: Scala 3 in sbt 1.5
+---
+
+We are delighted to announce that sbt 1.5.0 supports Scala 3 out-of-the-box.
+
+From the sbt user viewpoint, Scala 3 is just another Scala version, since every default task and setting works the same.
+
+Yet Scala 3 changes the picture of compatibility in the ecosystem and is going to change our way of dealing with Scala versions and library dependencies.
+That new reality is reflected in sbt by the `scalaBinaryVersion` setting and by the new `CrossVersion.for3Use2_13` and `CrossVersion.for2_13Use3`.
+
+Scala 3 early adopters, we encourage you to jump to sbt 1.5.0 and to drop the soon-to-be deprecated sbt-dotty plugin.
+
+## Just another Scala version
+
+Using Scala 3 in sbt is as short as:
+
+```scala
+// project/build.properties
+sbt.version=1.5.0
+```
+
+```scala
+// build.sbt
+scalaVersion := "3.0.0-RC2"
+```
+
+All the default tasks and settings are supported.
+You can manage dependencies, compile, test, run, package and publish your program or library. 
+
+Most sbt plugins should work the same, as long as they do not rely on a Scala 2 compiler plugin or a Scala 2-specific option.
+
+### The doc tasks
+
+On a Scala 3 project, the `doc` task invokes the [new Scaladoc tool](https://medium.com/virtuslab/the-future-of-scaladoc-8bf78a4e2988) which uses TASTy files to generate the documentation.
+
+For your own curiosity, you can run the `show tastyFiles` command to see the TASTy files of your Scala 3 project.
+
+### Porting a Scala 2 project
+
+You can change the `scalaVersion` of your project or you can add Scala 3 to the `crossScalaVersions` setting.
+
+```scala
+// build.sbt
+crossScalaVersions := Seq(“2.13.5”, “3.0.0-RC2”)
+```
+
+It is likely that you need to upgrade some dependencies, to change some compiler options or to remove some compiler plugins. Read the [Scala 3 migration tutorial](https://scalacenter.github.io/scala-3-migration-guide/docs/tutorials/sbt-migration.html) to learn more about those steps.
+
+## A new compatibility era
+
+Scala 3 introduces TASTy, an intermediate representation of Scala programs, that is designed for stability during the lifetime of Scala 3.
+The current state of TASTy makes us confident that all Scala 3 minor versions are going to be **backward binary compatible**.
+
+In other words, your Scala 3.x project can depend on libraries compiled with Scala 3.y, where y is lower or equal to x.
+
+This is materialized in sbt by the `scalaBinaryVersion` setting which is set to “3” on all Scala 3 versions (starting from 3.0.0).
+
+Let's take an example:
+
+```scala
+// build.sbt
+scalaVersion := "3.0.0"
+
+organization := "my-org"
+name := "my-project"
+version := "1.0.0"
+
+libraryDependency += "org.typelevel" %% "cats-core" % "x.y.z"
```

Comment:
I suggest omitting the settings that are not relevant for what you want to show.

```suggestion
scalaVersion := "3.0.0-RC2"
libraryDependencies += "org.typelevel" %% "cats-core" % "x.y.z"
```